src/status.rs

```diff
@@ -1,167 +1,162 @@
-use std::fmt;
-
+use bytes::Bytes;
 use h2;
-use http::header::{HeaderMap, HeaderValue};
-use http::status::StatusCode;
+use http::{self, HeaderMap};
+use http::header::{HeaderValue, InvalidHeaderValueBytes};
+use std::io;
 
 #[derive(Debug, Clone)]
 pub struct Status {
     code: Code,
+    error_message: Option<Bytes>,
+    binary_error_details: Option<Bytes>,
 }
 
-#[derive(Clone, Copy, PartialEq, Eq)]
-pub struct Code(Code_);
+const GRPC_STATUS_HEADER_CODE: &str = "grpc-status";
+const GRPC_STATUS_MESSAGE_HEADER: &str = "grpc-message";
+const GRPC_STATUS_DETAILS_HEADER: &str = "grpc-status-details-bin";
 
 impl Status {
-    #[inline]
-    pub fn code(&self) -> Code {
-        self.code
+    pub fn with_code(code: Code) -> Status {
+        Status {
+            code,
+            error_message: None,
+            binary_error_details: None,
+        }
     }
 
-    pub const OK: Status = Status {
```

Comment:
What was the reason to remove this? It seems convenient as a way to get quick status codes?

Reply:
`Code` is now a public enum, so can easily be gotten as needed